misc-scripts/xref_mapping/XrefParser/MGI_CCDS_Parser.pm

```diff
@@ -23,126 +24,143 @@ use strict;
-  while (my @row = $sth->fetchrow_array()) {
-    $ccds_label_to_xref_id{$row[0]} = $row[1];
+  my ($xref_label, $xref_id);
+  $sth->bind_columns( \$xref_label, \$xref_id );
+
+  while ( $sth->fetch() ) {
+    $ccds_label_to_xref_id{ $xref_label } = $xref_id;
   }
   $sth->finish;
 
-
-
-  my $ua = LWP::UserAgent->new();
-  $ua->timeout(10);
-  $ua->env_proxy();
-  
-
-  my $count = 0;
-  my $ccds_missing = 0;
+  my $count              = 0;
+  my $ccds_missing       = 0;
   my $entrezgene_missing = 0;
 
   my $mgi_io = $self->get_filehandle($file);
   if ( !defined $mgi_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
+    croak "Could not open $file\n";
   }
 
-    #
-    #
-    ##chromosome	g_accession	gene	gene_id	ccds_id	ccds_status	cds_strand	cds_from	cds_to	cds_locations	match_type
-    #1	NC_000067.5	Xkr4	497097	CCDS14803.1	Public	-	3206102	3661428	[3206102-3207048, 3411782-3411981, 3660632-3661428]	Identical
-    #1	NC_000067.5	Rp1h	19888	CCDS14804.1	Public	-	4334680	4342905	[4334680-4340171, 4341990-4342161, 4342282-4342905]	Identical
-    while (my $line = $mgi_io->getline()) {
-      my($chrom, $g_acc, $gene_name, $entrez_id, $ccds, @junk) = split(/\t/,$line);
-      if(defined($ccds_label_to_xref_id{$ccds})){ 
-	if(defined($accession{$gene_name}) and
-	   defined($label{$accession{$gene_name}})){
-	  my $acc = $accession{$gene_name};
-	  $self->add_dependent_xref({ master_xref_id => $ccds_label_to_xref_id{$ccds},
-				      acc            => $acc,
-				      version        => $version{$acc},
-				      label          => $label{$acc},
-				      desc           => $description{$acc},
-				      source_id      => $source_id,
-                                      dbi            => $dbi,
-				      species_id     => $species_id });
-
-	  $count++;
-	}
-	else{
-	  $entrezgene_missing++;
-	}
+  #
+  #
+  ##chromosome	g_accession	gene	gene_id	ccds_id	ccds_status	cds_strand	cds_from	cds_to	cds_locations	match_type
+  #1	NC_000067.5	Xkr4	497097	CCDS14803.1	Public	-	3206102	3661428	[3206102-3207048, 3411782-3411981, 3660632-3661428]	Identical
+  #1	NC_000067.5	Rp1h	19888	CCDS14804.1	Public	-	4334680	4342905	[4334680-4340171, 4341990-4342161, 4342282-4342905]	Identical
+  my $input_file = Text::CSV->new(
+    {
+      sep_char           => "\t",
+      empty_is_undef     => 1,
+      strict             => 1,
+      allow_loose_quotes => 1,
+    }
+  ) or croak "Cannot use file $file: " . Text::CSV->error_diag();
+
+  $input_file->column_names( [qw( chromosome g_accession gene gene_id ccds_id)] );    #ignore the rest of the columns
+
+  while ( my $data = $input_file->getline_hr($mgi_io) ) {
+    my $ccds      = $data->{'ccds_id'};
+    my $gene_name = $data->{'gene'};
+    if ( defined( $ccds_label_to_xref_id{$ccds} ) ) {
+      if (  defined( $accession{$gene_name} )
+        and defined( $label{ $accession{$gene_name} } ) )
+      {
+        my $acc = $accession{$gene_name};
```

Comment:
$acc is previously declared on line 72.
It's a good idea not to reuse variable names in the same lexical scope, consider renaming it.